internal/pkg/cli/deploy/deploy.go

```diff
@@ -0,0 +1,873 @@
+			appVersionGetter VersionGetter
+		)
+		if appVersionGetter, err = in.NewAppVersionGetter(d.App.Name); err != nil {
+			return nil, err
+		}
+
+		if err = validateRDSvcAliasAndAppVersion(d.Name,
+			aws.StringValue(t.Alias), d.Env.Name, d.App, appVersionGetter); err != nil {
+			return nil, err
+		}
+
+		if urls, err = uploadRDWSCustomResources(in.UploadOpts, d.S3Bucket); err != nil {
+			return nil, err
+		}
+		output.conf, err = stack.NewRequestDrivenWebServiceWithAlias(t, d.Env.Name, appInfo, *rc, urls)
+	case *manifest.BackendService:
+		output.conf, err = stack.NewBackendService(t, d.Env.Name, d.App.Name, *rc)
+	case *manifest.WorkerService:
+		var topics []deploy.Topic
+		topics, err = in.SNSTopicsLister.ListSNSTopics(d.App.Name, d.Env.Name)
+		if err != nil {
+			return nil, fmt.Errorf("get SNS topics for app %s and environment %s: %w", d.App.Name, d.Env.Name, err)
+		}
+		var topicARNs []string
+		for _, topic := range topics {
+			topicARNs = append(topicARNs, topic.ARN())
+		}
+		type subscriptions interface {
+			Subscriptions() []manifest.TopicSubscription
+		}
+
+		subscriptionGetter, ok := mft.(subscriptions)
+		if !ok {
+			return nil, errors.New("manifest does not have required method Subscriptions")
+		}
+		// Cache the subscriptions for later.
+		output.subscriptions = subscriptionGetter.Subscriptions()
+
+		if err = validateTopicsExist(output.subscriptions, topicARNs, d.App.Name, d.Env.Name); err != nil {
+			return nil, err
+		}
+		output.conf, err = stack.NewWorkerService(t, d.Env.Name, d.App.Name, *rc)
+
+	default:
+		return nil, fmt.Errorf("unknown manifest type %T while creating the CloudFormation stack", t)
+	}
+	if err != nil {
+		return nil, fmt.Errorf("create stack configuration: %w", err)
+	}
+	return &output, nil
+}
+
+func buildArgs(name, imageTag, workspacePath string, unmarshaledManifest interface{}) (*dockerengine.BuildArguments, error) {
+	type dfArgs interface {
+		BuildArgs(rootDirectory string) *manifest.DockerBuildArgs
+		ContainerPlatform() string
+	}
+	mf, ok := unmarshaledManifest.(dfArgs)
+	if !ok {
+		return nil, fmt.Errorf("%s does not have required methods BuildArgs() and ContainerPlatform()", name)
+	}
+	var tags []string
+	if imageTag != "" {
+		tags = append(tags, imageTag)
+	}
+	args := mf.BuildArgs(workspacePath)
+	return &dockerengine.BuildArguments{
+		Dockerfile: *args.Dockerfile,
+		Context:    *args.Context,
+		Args:       args.Args,
+		CacheFrom:  args.CacheFrom,
+		Target:     aws.StringValue(args.Target),
+		Platform:   mf.ContainerPlatform(),
+		Tags:       tags,
+	}, nil
+}
+
+func envFile(unmarshaledManifest interface{}) string {
+	type envFile interface {
+		EnvFile() string
+	}
+	mf, ok := unmarshaledManifest.(envFile)
+	if ok {
+		return mf.EnvFile()
+	}
+	// If the manifest type doesn't support envFiles, ignore and move forward.
+	return ""
+}
+
+type UploadCustomResourcesOpts struct {
+	uploader      CustomResourcesUploader
+	newS3Uploader func() (Uploader, error)
```

Comment:
If these fields are private how will the client set them 🤔 ?  
My preference is for the `cli/deploy` package not to have any `newXXXX` fields to create clients. Instead, we should have dependency injection from the `cli` package.
1. The `cli` package is responsible for creating these objects
2. The `cli/deploy` package accepts only interfaces or data types:   
```suggestion
	CustomResources      CustomResourcesUploader
	S3                   Uploader
```